src/components/PopupWithForm.js

```diff
@@ -6,6 +6,11 @@ class PopupWithForm extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
     this._inputList = this._popupElement.querySelectorAll(".modal__input");
+    this._saveButton = this._popupElement.querySelector(".modal__button");
+  }
+
+  setSubmitAction(handleFormSubmit) {
```

Comment:
Needs correction: you don't need this method here, you should create a different class `PopupWithConfirmation` for the popup with confirmation.

Reply:
Needs correction: please remove this function, you don't need it in this class